File: Leetcode/Java/Easy/LongestCommonPrefix.java

class Solution {
    public String longestCommonPrefix(String[] strs) {
      String first_word =  strs[0];//Gets the first word
        for (int i = 1; i < strs.length; i++) {
            String s = strs[i];
            while (!s.startsWith(first_word)){
                first_word = first_word.substring(0, first_word.length() - 1);
                if (first_word.isEmpty())
                    return "";
            }
        }
        return first_word;
    }
}
//Accepted answer

File: Leetcode/PythonAnswers/Easy/LongestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        first_word = strs[0]
        for s in strs[1:]:
            while not s.startswith(first_word):
                first_word =first_word[:-1]
                if not first_word:
                    return ""
                

        return first_word

#Take the first word and compare it to the rest checking first letter against first 
# second against second and so on until not a match

#Accept python answer